### PR TITLE
teal.data - bs345 - fluidPage

### PR DESCRIPTION
**Pull request by Polkas**
linked to https://github.com/insightsengineering/teal/issues/719

When shiny app theme is updated all fluidPage  inside another fluidPage should to have the same theme.

### CODE REVIEW

File: vignettes/using-delayed-data-advanced.Rmd [hunk 1]
```diff
@@ -576,6 +576,7 @@ data <- teal_data(adsl, adrs, x)
 shinyApp(
   ui = fluidPage(
     shinyjs::useShinyjs(),
+    theme = get_teal_bs_version(),
```
Comment (Polkas) [inline vignettes/using-delayed-data-advanced.Rmd:579]:
```suggestion
```

File: R/utils.R [hunk 2]
```diff
@@ -136,3 +136,15 @@ validate_metadata <- function(metadata) {
...
+get_teal_bs_version <- function() {
+  bs_theme <- getOption("teal.bs_theme")
+  if (is.null(bs_theme) || bs_theme == "") return(NULL)
+  if (!inherits(bs_theme, "bs_theme")) {
+    warning("teal.bs_theme has to be of a bslib::bs_theme class, the bslib::bs_theme(version = '3') is used.")
+    bslib::bs_theme(version = "3")
+  } else {
+    bs_theme
+  }
+}
```
Comment (Polkas) [inline R/utils.R:150]:
```suggestion
#' Resolve the expected bootstrap version
#' @keywords internal
get_teal_bs_version <- function() {
  bs_theme <- getOption("teal.bs_theme")
  if (is.null(bs_theme)) {
    NULL
  } else if (!inherits(bs_theme, "bs_theme")) {
    warning("teal.bs_theme has to be of a bslib::bs_theme class, the default shiny bootstrap is used.")
    NULL
  } else {
    bs_theme
  }
}
```

File: DESCRIPTION [hunk 3]
```diff
@@ -19,6 +19,7 @@ Depends:
     R (>= 4.0),
     shiny
 Imports:
+    bslib,
```
Comment (Polkas) [inline DESCRIPTION:22]:
```suggestion
```

File: R/utils.R [hunk 4]
```diff
@@ -136,3 +136,16 @@ validate_metadata <- function(metadata) {
   })
   return(NULL)
 }
+#' Resolve the expected bootstrap version
```
Comment (Polkas) [inline R/utils.R:139]:
```suggestion

#' Resolve the expected bootstrap version
```

File: R/utils.R [hunk 5]
```diff
@@ -136,3 +136,17 @@ validate_metadata <- function(metadata) {
   })
   return(NULL)
 }
+
```
Comment (Polkas) [inline R/utils.R:139]:
```suggestion


```

**Review by mhallal1 — APPROVED** [review]
LGTM

**Comment by asbates** [comment]
LGTM

**Comment by donyunardi** [comment]
The button lost its background color for selected theme:

### 1. TealDataConnector 
with `options("teal.bs_theme" = bslib::bs_theme(version = 5, bootswatch = "lux"))` setup.

The button looks fine when run using other bootswatch themes.

### 2. TealDatasetConnector 
with `options("teal.bs_theme" = bslib::bs_theme(version = 5))` setup.

The button looks fine if I pass a bootswatch theme.

These could be minor issue (theme related).
Everything else looks good to me.

**Review by donyunardi — APPROVED** [review]

**Pull request merged**